packages/connect-core/src/connect-error.ts

```diff
@@ -65,6 +65,13 @@ export class ConnectError extends Error {
 
   override name = "ConnectError";
 
+  /**
+   * The underlying cause of this error.  In cases where the actual cause is
+   * elided with the error message, the cause is specified here so that we don't
+   * leak the underlying error, but instead make it available for logging.
+   */
+  cause: unknown;
```

Comment:
Can you make this a `unknown | undefined`, so that it is obvious that it is optional?

Reply:
Do you think it makes more sense to make this a union type of like `Error | undefined | blah`?

Reply:
Unknown is the best choice. In current TypeScript, `e` in `try{}catch(e){}` is `unknown`.